dataflowAPI/rose/x86_64InstructionSemantics.h

```diff
@@ -16,13 +16,14 @@
 #include "SgAsmExpression.h"
 #include "conversions.h"
 
-#ifndef ROSE_X86INSTRUCTIONSEMANTICS_H
-
+#undef ROSE_ASSERT
 #define ROSE_ASSERT(x) \
     if (!(x)) {\
         throw rose::BinaryAnalysis::InstructionSemantics2::BaseSemantics::Exception("", NULL); \
     }\
 
+#ifndef ROSE_X86INSTRUCTIONSEMANTICS_H
```

Comment:
@mxz297 I realize this was the case before this change, but why is the file guard not guarding the entire file?

Reply:
@hainest I have no clue. This is a legacy source file from the ROSE project...

Reply:
@mxz297 Neat

@stanfordcox Is there a specific reason for moving the definition of ROSE_ASSERT outside the file guard?

Reply:
@hainest To be clear, even though this is a legacy file, I add the ROSE_ASSERT macro so that it will throw an exception, instead of asserting. This worked fine for me.

However, this does not work for Stan as Stan explained that what I have written may lead to a macro-redefinition. So, by moving the macro outside of the file guard, it will force the macro to be effective for this file, and ensure that ROSE_ASSERT throws an exception instead of asserting.

Reply:
SymbolicExpansion.C includes
 /home/scox/dyninst/src/dataflowAPI/src/../rose/x86InstructionSemantics.h
  => #define ROSE_X86_64INSTRUCTIONSEMANTICS_H includes
  =>./home/scox/dyninst/src/dataflowAPI/src/../rose/rose.h
       => #define ROSE_ASSERT assert
/home/scox/dyninst/src/dataflowAPI/src/../rose/x86_64InstructionSemantics.h
  => #ifndef ROSE_X86INSTRUCTIONSEMANTICS_H
  => #define ROSE_ASSERT(x)
  => getSegregFromMemoryReference
  => #endif
 does the ifndef block ever get invoked?  It seems everything SymbolicExpansion.C, SymVal.C includes both such that ROSE_X86INSTRUCTIONSEMANTICS_H is always defined

Reply:
awkward last sentence:  the files that include x86_64InstructionSemantics.h also include x86InstructionSemantics.h:  SymbolicExpansion.C, SymVal.C.   So I'm wondering if the block at #ifndef ROSE_X86INSTRUCTIONSEMANTICS_H is actually ever invoked since x86InstructionSemantics.h defines ROSE_X86INSTRUCTIONSEMANTICS_H